contracts/mainnet/connectors/compound/v3/main.sol

```diff
@@ -0,0 +1,845 @@
+				from: from,
+				to: to,
+				amt: amt,
+				getId: getId,
+				setId: setId
+			})
+		);
+
+		eventName_ = "LogWithdrawFromUsingManager(address,address,address,address,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(market, token, from, to, amt_, getId, setId_);
+	}
+
+	/**
+	 * @dev Borrow base asset.
+	 * @notice Borrow base token from Compound.
+	 * @param market The address of the market.
+	 * @param token The address of the token to be borrowed. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param amt The amount of base token to borrow.
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens borrowed.
+	 */
+	function borrow(
+		address market,
+		address token,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		uint256 amt_ = getUint(getId, amt);
+
+		require(market != address(0), "invalid market address");
+
+		bool isEth = token == ethAddr;
+		address token_ = getBaseToken(market);
+		require(token == token_ || token == ethAddr, "invalid-token");
+
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		if (token_ == getBaseToken(market)) {
+			uint256 balance = CometInterface(market).balanceOf(address(this));
+			require(balance == 0, "borrow-disabled-when-supplied-base");
+		}
+
+		uint256 initialBal = CometInterface(market).borrowBalanceOf(
+			address(this)
+		);
+
+		CometInterface(market).withdraw(token_, amt_);
+
+		uint256 finalBal = CometInterface(market).borrowBalanceOf(
+			address(this)
+		);
+
+		amt_ = sub(finalBal, initialBal);
+
+		convertWethToEth(isEth, tokenContract, amt_);
+
+		setUint(setId, amt_);
+
+		eventName_ = "LogBorrow(address,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(market, amt_, getId, setId);
+	}
+
+	/**
+	 * @dev Borrow base asset and transfer to 'to' account.
+	 * @notice Borrow base token from Compound on behalf of an address.
+	 * @param market The address of the market.
+	 * @param token The address of the token to be borrowed. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param to The address to which the borrowed asset is transferred.
+	 * @param amt The amount of the token to withdraw. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens borrowed.
+	 */
+	function borrowTo(
+		address market,
+		address token,
+		address to,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		require(
+			token == ethAddr || token == getBaseToken(market),
+			"invalid-token"
+		);
+		(uint256 amt_, uint256 setId_) = _borrow(
+			BorrowWithdrawParams({
+				market: market,
+				token: token,
+				from: address(0),
+				to: to,
+				amt: amt,
+				getId: getId,
+				setId: setId
+			})
+		);
+		eventName_ = "LogBorrowOnBehalf(address,address,uint256,uint256,uint256)";
```

Comment:
@Richa-iitr We need to change the event name too

Reply:
done